org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TaskViewContentProvider.java

```diff
@@ -131,11 +142,34 @@ public boolean hasChildren(Object element) {
     }
 
     private Object[] childrenOf(ProjectNode projectNode) {
+        boolean isGroupProjects = this.taskView.getState().isGroupProjects();
```

Comment:
Please clean this up a bit. This method should look something like this: 
    
    if (this.taskView.getState().isGroupProjects()) {
        result.addAll(projectNodesFor(projectNode));
    }
    if (this.taskView.getState().isGroupTasks()) {
            result.addAll(groupNodesFor(projectNode));
    } else {
        result.addAll(taskNodesFor(projectNode));
    }

The `projectNodesFor()` method should look very similar to `taskNodesFor()`.